compiler/include/baseAST.h

```diff
@@ -543,6 +547,17 @@ static inline const CallExpr* toConstCallExpr(const BaseAST* a)
   case E_TryStmt:                                                       \
     AST_CALL_CHILD(_a, TryStmt, body(), call, __VA_ARGS__);             \
     break;                                                              \
+  case E_ForallStmt: {                                                  \
+    AST_CALL_LIST (_a, ForallStmt,    iterExprs(),  call, __VA_ARGS__); \
+    ForallIntents* fi = ((ForallStmt*)_a)->withClause();                \
+    AST_CALL_STDVEC(fi->fiVars,  Expr,              call, __VA_ARGS__); \
+    AST_CALL_STDVEC(fi->riSpecs, Expr,              call, __VA_ARGS__); \
+    AST_CALL_CHILD(fi, ForallIntents, iterRec,      call, __VA_ARGS__); \
+    AST_CALL_CHILD(fi, ForallIntents, leadIdx,      call, __VA_ARGS__); \
+    AST_CALL_CHILD(fi, ForallIntents, leadIdxCopy,  call, __VA_ARGS__); \
+    AST_CALL_CHILD(_a, ForallStmt,    masterBlock(),call, __VA_ARGS__); \
```

Comment:
I think it is deceptive to use methods like iterExprs() or masterBlock() here. Basically, this section of code needs to track any modifications to the internal representation of ForallStmt. So, I'd personally make those fields public and just have private-looking names (e.g. _masterBlock). But that is up to you.

Arguably we all should just know that this code is sensitive to representational changes...